docs/architecture/cryptography/crypto-guide.md

```diff
@@ -0,0 +1,186 @@
+---
+title: How to use Cryptography in Bitwarden
+sidebar_label: Cryptography Guide
+description:
+  A guide on using cryptographic APIs within Bitwarden, targeted at non-cryptography software
+  engineers.
+---
+
+# How to use cryptography in Bitwarden
+
+This guide is aimed at non-cryptography teams that want to consume cryptographic APIs in order to
+build features that need end-to-end encryption. This guide provides an overview of available tools
+for composing features that need cryptographic protection.
+
+Currently, there is a set of low-level APIs
+([EncString](https://github.com/bitwarden/sdk-internal/blob/c60a5d794732d2c0fc203feb21ce5851d5325fe1/crates/bitwarden-crypto/src/enc_string/symmetric.rs#L59),
+[UnsignedSharedKey](https://github.com/bitwarden/sdk-internal/blob/c60a5d794732d2c0fc203feb21ce5851d5325fe1/crates/bitwarden-crypto/src/enc_string/asymmetric.rs#L58),
+[MasterKey](https://github.com/bitwarden/sdk-internal/blob/c60a5d794732d2c0fc203feb21ce5851d5325fe1/crates/bitwarden-crypto/src/keys/master_key.rs#L32))
+that have been used to build most features, with each team owning the cryptographic constructions
+created. Recently, high-level safe primitives were introduced that will move the complexity out of
+each team's ownership. These are not yet complete, and if a particular use-case is not covered by
+them, teams should reach out to the key-management team! The goal of these is to have most teams
+never have to think about cryptography, or having to do safety analysis, or to own any cryptographic
+construct or protocol. These high-level primitives abstract away all complex details and give teams
+a low-complexity, easy to use and hard to misuse interface to work with.
+
+Primarily, this is aimed for consumption for end-to-end encrypted storage of long-term data, in
+products such as the password manager or secrets manager.
+
+## Rules
+
+The primary rule here is: don't roll your own cryptography. Where possible, high level, safe tested
+and analyzed protocols and primitives need to be used. The higher level the primitive, the less
+likely that security bugs get introduced, and the less complexity to maintain and keep track of.
+Only where not otherwise possible should low level primitives be used, and this should be done with
+extreme caution and oversight.
+
+Encryption in the TypeScript clients for new cases is deprecated. Any new cryptographic code must be
+written in the SDK if possible. Existing use-cases can be continued in the TypeScript clients for
+now, but eventually will be migrated too. There are several reasons behind this. On the one hand the
+SDK has better memory safety guarantees and prevents key material from being left behind in memory.
+On the other hand, newer, safer APIs are not exposed outside of the SDK.
```

Comment:
This hand metaphor typically is used to show pros and cons, but these are both positive.

Please don't accept this suggestion, I suspect prettier will want to reformat

```suggestion
now, but eventually will be migrated too. First, the
SDK has better memory safety guarantees and prevents key material from being left behind in memory.
Second, newer, safer APIs are not exposed outside of the SDK.
```